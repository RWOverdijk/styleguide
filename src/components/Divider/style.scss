@import "../../css/_global.scss";

.Van-Divider {
  position: relative;
  width: 100%;
  height: 1px;
  background-color: #f2f2f2;
  display: block;
  margin: 20px 0;
  text-align: center;

  &-content {
    position: relative;
    top: -10px;
    background-color: $color-white;
    display: inline-block;
    padding: 0 10px;
    color: $color-primary-brightst;
    font-size: 14px;
    font-weight: $font-weight-semibold;
  }
}
