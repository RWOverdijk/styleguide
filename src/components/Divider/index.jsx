import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './style.css';

class Divider extends PureComponent {
  getClass(className) {
    let classes = ['Van-Divider'];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    classes.push(className);

    return classes.join(' ');
  }

  render() {
    return <span className={this.getClass()}>
      <span className="Van-Divider-content">{this.props.children}</span>
    </span>;
  }
}

Divider.propTypes = {
  className: PropTypes.string
};

export default Divider;
