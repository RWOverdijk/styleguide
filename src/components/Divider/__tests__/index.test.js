import React from 'react';
import renderer from 'react-test-renderer';
import Divider from '../';

describe('<Divider>', () => {
  it("renders Divider", () => {
    const tree = renderer.create(
      <Divider />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Divider with content", () => {
    const tree = renderer.create(
      <Divider>Hi</Divider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Divider className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Divider data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
