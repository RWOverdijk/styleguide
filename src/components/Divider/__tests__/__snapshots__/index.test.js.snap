// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider> renders Divider 1`] = `
<span
  className="Van-Divider"
>
  <span
    className="Van-Divider-content"
  >
    
  </span>
</span>
`;

exports[`<Divider> renders Divider with content 1`] = `
<span
  className="Van-Divider"
>
  <span
    className="Van-Divider-content"
  >
    Hi
  </span>
</span>
`;

exports[`<Divider> renders vertical Divider 1`] = `
<span
  className="Van-Divider Van-Divider--vertical"
>
  <span
    className="Van-Divider-content"
  >
    
  </span>
</span>
`;

exports[`<Divider> renders with specific attributes 1`] = `
<span
  className="Van-Divider"
  data-id="0"
>
  <span
    className="Van-Divider-content"
  >
    
  </span>
</span>
`;

exports[`<Divider> renders with specific className 1`] = `
<span
  className="Van-Divider my-class"
>
  <span
    className="Van-Divider-content"
  >
    
  </span>
</span>
`;
